### DISCUSSION TITLE
Run autouse fixture at the end of fixtures declaration

### DISCUSSION THREAD

**Discussion by victormaricato** [pytest-dev/pytest#8572] [Q&A]
Let's say I have the following fixtures:

```
@fixture
def does_something_to_db():
     # Something to DB
     
@fixture(autouse=True)
def commit_changes_to_db(db):
     db.commit()
```

How can I make `commit_changes_to_db` run everytime that `does_something_to_db` runs?

With the constraint that I cannot pass `does_something_to_db` as an argument to `commit_changes_to_db` (declare dependency). This constraint is due to a lot of fixtures requiring `commit_changes_to_db` at the end. Therefore, the argument list would be unnecessarily (because the fixtures are not used inside it) enormous.

**Comment by The-Compiler** [comment]
> With the constraint that I cannot pass `does_something_to_db` as an argument to `commit_changes_to_db` (declare dependency). This constraint is due to a lot of fixtures requiring `commit_changes_to_db` at the end. Therefore, the argument list would be unnecessarily (because the fixtures are not used inside it) enormous.

I don't understand what you mean. Do you mean the other way around, i.e. `def does_something_to_db(commit_changes_to_db):`, but you have many fixtures which would require adding `commit_changes_to_db`? I don't really see a way around that.

**Reply by victormaricato** [reply]
The case you mentioned would be ok to read, therefore I would definitely consider using it, however, the syntax:
`def does_something_to_db(commit_changes_to_db)`
Makes `commit_changes_to_db` run BEFORE `does_something_to_db` not after, as expected, or am I missing something?

**Reply by The-Compiler** [reply]
A fixture [can also run teardown steps](https://docs.pytest.org/en/stable/fixture.html#teardown-cleanup-aka-fixture-finalization), so you'd need to do something like:

```python
def commit_changes_to_db(db):
    yield
    db.commit()
```
